sechub-testframework/src/main/java/com/daimler/sechub/test/PojoTester.java

```diff
@@ -10,92 +10,136 @@
+                    Type generic = readmethod.getGenericReturnType();
+                    ParameterizedType pm = (ParameterizedType) generic;
+                    Type[] args = pm.getActualTypeArguments();
+                    String typeName = args[0].getTypeName();
+                    type = Class.forName(typeName);
+                }
+                Object mockedArgument = null;
+                if (type.isPrimitive()) {
+                    if (type.equals(int.class)) {
+                        mockedArgument = 42;
+                    } else if (type.equals(double.class)) {
+                        mockedArgument = (float) 42;
+                    } else if (type.equals(byte.class)) {
+                        mockedArgument = (byte) 42;
+                    }
+                } else if (type.isEnum()) {
+                    Object[] enumObjects = type.getEnumConstants();
+                    mockedArgument = enumObjects[0];
+                } else {
+                    try {
+                        Constructor<?> constructor = type.getConstructor();
+                        mockedArgument = constructor.newInstance();
+                    } catch (Exception e) {
+                        /* no default constructor available - fall back to Mockito ... */
+                        try {
+                            mockedArgument = Mockito.mock(type);
+                        } catch (Exception e2) {
+                            throw new IllegalStateException("No default constructor available for:" + type + "\nat method:" + readmethod
+                                    + "\nDid try to creeate with mockito, but failed", e);
+                        }
+                    }
+                }
+                if (optional) {
+                    mockedArgument = Optional.of(mockedArgument);
+                }
+                try {
+                    writeMethod.invoke(objectToTest, mockedArgument);
+                } catch (Exception e) {
+                    throw new IllegalStateException(
+                            "Was not able to set mockedArgument:" + mockedArgument + "\ninto:" + objectToTest + "\nby writeMethod:" + writeMethod);
+                }
+
+                Object result = readmethod.invoke(objectToTest);
+
+                if (!Objects.equals(result, mockedArgument)) {
+                    fail("The getter/setter implementation of " + objectToTest.getClass() + " failed!\nProperty:" + descriptor.getName() + "\nExpected:"
+                            + mockedArgument + "\nGot:" + result);
+                }
+
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            fail("Bean introspection failed - internal error in framework. See output for details - message was:" + e.getMessage());
+        }
+    }
+
+    public interface PojoChanger<T> {
+        void changePojo(T pojo);
+    }
+
+    /**
+     * Change a POJO. This is a handy method for POJOs having setters but no fluent
```

Comment:
Please rephrase this sentence it is not understandable.

Reply:
The complete phrase was:
```
Change a POJO. This is a handy method for POJOs having setters but no fluent  API but a need in unit tests.

Example:
<pre>
      change(createExample(), (example) -> example.setName("changed...") ));
</pre>
 ```
when you look at the call hierarchy of the method it is very often used inside test closures.

I have changed the javadoc now as following
```
Changes given POJO by given POJO changer. Use this method when you are
writing tests and you need a fluent API but the POJO does not provide this.
Look at the next example or look into call hierarchy for more details.<br>

Example:

<pre>
     testBothAreNOTEqual(createExample(), change(createExample(), (codeFlow) -> codeFlow.setMessage(new Message("other"))));
</pre>
```

Reply:
done